src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -100,5 +265,18 @@ protected virtual string CreateDescription(OperationGroup operationGroup, string
                 $"A class representing the operations that can be performed over a specific {clientPrefix}." :
                 BuilderHelpers.EscapeXmlDescription(operationGroup.Language.Default.Description);
         }
+
+        public class ResourceListMethod
+        {
+            public PagingMethod? PagingMethod { get; }
+
+            public ClientMethod? ClientMethod { get; }
+
+            public ResourceListMethod(PagingMethod? pagingMethod = null, ClientMethod? clientMethod = null)
+            {
+                PagingMethod = pagingMethod;
+                ClientMethod = clientMethod;
+            }
```

Comment:
Since we should always only assign one of these two (and never both `null`), maybe we need to introduce two extra constructor to assign only one of them each

I hope we could have union type in C# soon
```suggestion
            public ResourceListMethod(PagingMethod pagingMethod)
            {
                PagingMethod = pagingMethod;
                ClientMethod = null;
            }
            public ResourceListMethod(ClientMethod clientMethod)
            {
                PagingMethod = null;
                ClientMethod = clientMethod;
            }
```